java/system-test/src/main/java/sleeper/systemtest/compaction/WaitForCurrentSplitAddingMissingJobs.java

```diff
@@ -0,0 +1,75 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.systemtest.compaction;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sleeper.compaction.job.CompactionJobStatusStore;
+import sleeper.compaction.status.store.job.DynamoDBCompactionJobStatusStore;
+import sleeper.systemtest.SystemTestProperties;
+import sleeper.systemtest.util.InvokeLambda;
+
+import java.io.IOException;
+
+import static sleeper.configuration.properties.SystemDefinedInstanceProperty.COMPACTION_JOB_CREATION_LAMBDA_FUNCTION;
+import static sleeper.configuration.properties.SystemDefinedInstanceProperty.SPLITTING_COMPACTION_TASK_CREATION_LAMBDA_FUNCTION;
+
+public class WaitForCurrentSplitAddingMissingJobs {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WaitForCurrentSplitAddingMissingJobs.class);
+
+    private WaitForCurrentSplitAddingMissingJobs() {
+    }
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        if (args.length != 2) {
+            System.out.println("Usage: <instance id> <table name>");
+            return;
+        }
+
+        String instanceId = args[0];
+        String tableName = args[1];
+
+        AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+        AmazonDynamoDB dynamoDBClient = AmazonDynamoDBClientBuilder.defaultClient();
+        AmazonSQS sqsClient = AmazonSQSClientBuilder.defaultClient();
+
+        SystemTestProperties systemTestProperties = new SystemTestProperties();
+        systemTestProperties.loadFromS3GivenInstanceId(s3Client, instanceId);
+        CompactionJobStatusStore store = DynamoDBCompactionJobStatusStore.from(dynamoDBClient, systemTestProperties);
+
+        WaitForPartitionSplitting waitForSplitting = new WaitForPartitionSplitting(sqsClient, systemTestProperties);
+        WaitForCompactionJobs waitForCompaction = new WaitForCompactionJobs(store, tableName);
+
+        LOGGER.info("Waiting for partition splits");
+        waitForSplitting.pollUntilFinished();
+        LOGGER.info("Creating compaction jobs");
+        InvokeLambda.forInstance(instanceId, COMPACTION_JOB_CREATION_LAMBDA_FUNCTION);
+        if (store.getUnfinishedJobs(tableName).isEmpty()) {
+            LOGGER.info("Created no more jobs, splitting complete");
+            return;
+        }
+        LOGGER.info("Creating splitting compaction tasks");
+        InvokeLambda.forInstance(instanceId, SPLITTING_COMPACTION_TASK_CREATION_LAMBDA_FUNCTION);
```

Comment:
We have `WaitForQueueEstimateNotEmpty` polling in `SplitPartitionsUntilNoMoreSplits.java`. Do we need it here too?

Reply:
Done, thanks.